python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -1126,6 +1126,19 @@ def get_defer_args(self) -> Sequence[str]:
 
         return ["--defer", state_flag, self.state_path]
 
+    def get_state_args(self) -> Sequence[str]:
+        """Build the state arguments for the dbt CLI command, using the supplied state directory.
+        If no state directory is supplied, or the state directory does not have a manifest for.
+        comparison, an empty list of arguments is returned.
+
+        Returns:
+            Sequence[str]: The state arguments for the dbt CLI command.
+        """
+        if not (self.state_path and Path(self.state_path).joinpath("manifest.json").exists()):
+            return []
+
+        return ["--state", self.state_path]
```

Comment:
other options:
* just have users use the `state_path` property 
* add a `state_path_if_populated` 
* add a `get_state_path` that throws if manifest is missing 
* ???